### PR TITLE
Update MerlinAU.sh-Redirect-Unzip-Output

### PR DESCRIPTION
**Pull request by ExtremeFiretop**
Redirect Unzip Output to the logs.

### CODE REVIEW

**Comment by ExtremeFiretop** [comment]
Log file with this change:

> 2024-02-18 22:38:07 Required RAM: 227223 KB - RAM Free: 424952 KB - RAM Available: 520372 KB
> 2024-02-18 22:38:08 Backup Started (by BACKUPMON)
> 2024-02-18 22:38:53 Backup Finished
> 2024-02-18 22:38:53 Backup Completed Successfully
> 2024-02-18 22:38:53 Latest release version is 3004.388.6.0.
> 2024-02-18 22:38:53 Downloading https://sourceforge.net/projects/asuswrt-merlin/files/GT-AXE11000/Release/GT-AXE11000_3004_388.6_0.zip
> 2024-02-18 22:39:24 Required RAM: 227223 KB - RAM Free: 268756 KB - RAM Available: 517340 KB
> 2024-02-18 22:39:32 Archive:  /tmp/mnt/USB1/MerlinAU.d/GT-AXE11000_firmware/GT-AXE11000_firmware.zip
> 2024-02-18 22:39:32   inflating: GT-AXE11000_3004_388.6_0_pureubi.w
> 2024-02-18 22:39:32   inflating: Changelog-NG.txt
> 2024-02-18 22:39:32   inflating: sha256sum.sha256
> 2024-02-18 22:39:32   inflating: GT-AXE11000_3004_388.6_0_rog_pureubi.w
> 2024-02-18 22:39:32 Required RAM: 227223 KB - RAM Free: 99352 KB - RAM Available: 348408 KB
> 2024-02-18 22:39:32 No high-risk phrases found in the change-logs.
> 2024-02-18 22:39:32 Required RAM: 227223 KB - RAM Free: 99340 KB - RAM Available: 348396 KB
> 2024-02-18 22:39:32 ROG Build selected for flashing
> 2024-02-18 22:39:32 Required RAM: 227223 KB - RAM Free: 99492 KB - RAM Available: 348416 KB
> 2024-02-18 22:39:32 Router Web URL is: http://192.168.x.x

Log output without this change:

```
2024-02-18 22:48:28 Required RAM: 227223 KB - RAM Free: 424952 KB - RAM Available: 520372 KB
2024-02-18 22:48:38 Backup Started (by BACKUPMON)
2024-02-18 22:48:57 Backup Finished
2024-02-18 22:48:57 Backup Completed Successfully
2024-02-18 22:48:57 Latest release version is 3004.388.6.0.
2024-02-18 22:48:57 Downloading https://sourceforge.net/projects/asuswrt-merlin/files/GT-AXE11000/Release/GT-AXE11000_3004_388.6_0.zip
2024-02-18 22:49:45 Required RAM: 227223 KB - RAM Free: 268756 KB - RAM Available: 517340 KB
2024-02-18 22:49:47 Required RAM: 227223 KB - RAM Free: 99352 KB - RAM Available: 348408 KB
2024-02-18 22:49:47 No high-risk phrases found in the change-logs.
2024-02-18 22:49:47 Required RAM: 227223 KB - RAM Free: 99340 KB - RAM Available: 348396 KB
2024-02-18 22:49:47 ROG Build selected for flashing
2024-02-18 22:49:47 Required RAM: 227223 KB - RAM Free: 99492 KB - RAM Available: 348416 KB
2024-02-18 22:49:47 Router Web URL is: http://192.168.x.x
```

**Comment by ExtremeFiretop** [comment]
As a positive bonus it also redirects to the system logs too now:

image

File: MerlinAU.sh [hunk 1]
```diff
@@ -2788,8 +2788,10 @@ Please manually update to version $minimum_supported_version or higher to use th
     check_memory_and_prompt_reboot "$required_space_kb" "$availableRAM_kb"
 
     # Extracting the firmware binary image #
-    if unzip -o "$FW_ZIP_FPATH" -d "$FW_BIN_DIR" -x README*
-    then
+    if output=$(unzip -o "$FW_ZIP_FPATH" -d "$FW_BIN_DIR" -x README* 2>&1); then
+        echo "$output" | while IFS= read -r line; do
+            Say "$line"
+        done
```
Comment (Martinski4GitHub) [inline MerlinAU.sh:2794]: Very good idea. If something goes wrong at this step, it may help to have the output in the logs.

**Review by Martinski4GitHub — APPROVED** [review]

**Comment by Martinski4GitHub** [comment]
@ExtremeFiretop,
Don't forget to change the version string :>).

**Comment by ExtremeFiretop** [comment]
> @ExtremeFiretop, Don't forget to change the version string :>). Unless you plan to merge the change into the "master" branch as part of the latest release?

I forgot but then I headed out the door, this will be for the next release it's not mission critical it's just helpful like you said. If something goes wrong we'll at least see what other than our generic failed error message.

I'll be back in 30 minutes and I'll adjust it then, I literally hit commit as I was heading out the door 😔

**Comment by Martinski4GitHub** [comment]
> 
> I forgot but then I headed out the door, this will be for the next release it's not mission critical it's just helpful like you said. If something goes wrong we'll at least see what other than our generic failed error message.
> 
Ah OK. Yes, that makes sense.

**Comment by ExtremeFiretop** [comment]
> > I forgot but then I headed out the door, this will be for the next release it's not mission critical it's just helpful like you said. If something goes wrong we'll at least see what other than our generic failed error message.
> 
> Ah OK. Yes, that makes sense.

Done! :)

**Comment by Martinski4GitHub** [comment]
> 
> Done! :) (Sorry for the delay!)
>
No worries. We all have personal "stuff" to take care of & things to do... Life happens :>).

**Pull request merged**